### PR TITLE
Fixing issue with saving comments on staging

### LINKED ISSUE #93: Can't save candidate comments on staging
On the staging environment the save action on comments results in an unending http request.  No clue as to why, but it appears to be environment specific as it can't be replicated in dev

### PR DESCRIPTION
**Pull request by jackofseattle**
Fixes #93

The issue here was that the enum for comments.type only accepts 'contact' or 'server'; The front end was sending 'comment' which was making the update fail. We weren't catching the error and the request was never resolved.

This wasn't caught on dev machines because sqlite3 doesn't actually validate it's enum types.

### CODE REVIEW

File: server/routes/comments.js [hunk 1]
```diff
@@ -55,50 +42,49 @@ function validateBody(req, res, next) {
...
-          return res.status(404);
-        }
-        return res.json(result);
-      });
+  async getById(req, res) {
+    let result = await CommentModel
+      .findOne({where: {id: req.params.commentId}});
+
+    if(!result) {
+      return res.status(404);
```
Comment (daviskoh) [inline server/routes/comments.js]: Do we need to call `.send` or `.end` here?

Comment (jackofseattle) [inline server/routes/comments.js]: it seems to work as is.  But probably should.  To be fair, now that I think about it, I have no idea why you would ever need to fetch an individual comment anyways

File: client/vivace/comments/comment-form/controller.js [hunk 2]
```diff
@@ -11,7 +11,6 @@ module.exports = [
         this.newComment.contactMethod = this.newComment.otherMethod;
         delete this.newComment.otherMethod;
       }
-      this.newComment.type = 'comment';
```
Comment (daviskoh) [inline client/vivace/comments/comment-form/controller.js]: Do we then need to specify these as type `contact`?

File: server/routes/comments.js (hunk 1 above)
Comment (daviskoh) [inline server/routes/comments.js]: Indeed hence my comment / question from https://github.com/inrhythm/vivace/pull/94. I vote for removing it.

**Comment by daviskoh** [comment]
image

**Comment by jackofseattle** [comment]
Ignore coveralls - its showing the delta from the other PR branch (which hasn't been merged)

File: client/vivace/comments/comment-form/controller.js (hunk 2 above)
Comment (jackofseattle) [inline client/vivace/comments/comment-form/controller.js]: No - the default on the model is contact

**Comment by daviskoh** [comment]
Issue seems resolved for looks OK to merge.

***Unrelated issue*** is that when attempting to add new candidates, we seem to be posting to `/api/candidates/new`

image

**Comment by daviskoh** [comment]
:+1:

**Pull request merged**